contracts/BondFactory.sol

```diff
@@ -113,12 +122,20 @@ contract BondFactory is AccessControl {
         uint256 maxSupply
     ) external onlyIssuer returns (address clone) {
         clone = Clones.clone(tokenImplementation);
+
         isBond[clone] = true;
 
+        uint256 collateralDeposited = _deposit(
+            _msgSender(),
+            clone,
+            collateralToken,
+            maxSupply.mulDivUp(collateralRatio, ONE)
+        );
+
```

Comment:
Do we need to check the `collateralDeposited == maxSupply.mulDivUp(collateralRatio, ONE)` at this point to confirm the amount in the contract?

Reply:
That's a good question. What do you think about disallowing any fee tokens all together by doing the transfer then checking the balance of the bond contract and reverting if the number sent in does not match the amount in the bond contract?

Previously we couldn't do this because failing would put the bond in a bad step. Failing on the `createBond` state is fine.

Reply:
I'm going to do the above unless you stop me!

Reply:
fixed [88cadf4](https://github.com/porter-finance/v1-core/pull/175/commits/88cadf441a00e3bb1cfc4a940e2e78aee882a8b7)